### PR TITLE
AP_Scripting: cope better with high P gains in quicktune

### PR DESCRIPTION
**Pull request by tridge**
if we lower the D gain, then lower P and I by the same ratio before we
start on the P gain

Also added parameters to disable filter changes and control PI ratios

### CODE REVIEW

File: libraries/AP_Scripting/applets/VTOL-quicktune.lua [hunk 1]
```diff
@@ -283,8 +322,8 @@ end
 
 -- return gain limits on a parameter, or 0 for no limit
 function gain_limit(pname)
-   if string.sub(pname, 1, 3) == "YAW" then
-      local suffix = string.sub(pname, -2)
+   if param_axis(pname) == "YAW" then
+      local suffix = param_suffix(pname, -2)
```
Comment (hendjoshsr71) [inline libraries/AP_Scripting/applets/VTOL-quicktune.lua:326]: What does the extra argument `-2` do? 

```suggestion
      local suffix = param_suffix(pname, -2)
```

File: libraries/AP_Scripting/applets/VTOL-quicktune.lua [hunk 2]
```diff
@@ -244,12 +281,14 @@ function adjust_gain(pname, value)
...
-         I:set(value)
+
+      -- work out ratio of P to I that we want
+      local PI_ratio = QUIK_RP_PI_RATIO:get()
+      if param_axis(pname) == "YAW" then
+         PI_ratio = QUIK_Y_PI_RATIO:get()
+      end
+      if PI_ratio >= 1 then
+         I:set(value/PI_ratio)
       end
```
Comment (hendjoshsr71) [inline libraries/AP_Scripting/applets/VTOL-quicktune.lua:292]: If the ratio is less than 1 should we send a message or set P=I ?

Should we  prevent mistypes of ratios greater than a certain value or just warn?
```suggestion
      else
          I:set(value)
      end
```

**Review by hendjoshsr71 — COMMENTED** [review]
One unused  variable and question.

Tested the changes in real flight

File: libraries/AP_Scripting/applets/VTOL-quicktune.lua (hunk 1 above)
Comment (tridge) [reply libraries/AP_Scripting/applets/VTOL-quicktune.lua]: fixed, thanks

File: libraries/AP_Scripting/applets/VTOL-quicktune.lua (hunk 2 above)
Comment (tridge) [reply libraries/AP_Scripting/applets/VTOL-quicktune.lua:292]: i've added docs instead, saying it won't change i at all if the ratio is less than 1

**Review by hendjoshsr71 — APPROVED** [review]
lgtm

**Pull request merged**